### PR TITLE
Implement the patch method

### PR DESCRIPTION
**Pull request by dtaniwaki**
I made `patch` method which is easier to use than `jsonMergePatch`. It also supports different merge strategies supported by Kubernetes. Could you review and consider merging it?

### CODE REVIEW

**Comment by doriordan** [comment]
@dtaniwaki I have started reviewing this but am v. busy so it may be another couple of days before I complete the review - thanks for the PR.

File: client/src/main/scala/skuber/package.scala [hunk 1]
```diff
@@ -223,6 +224,65 @@ package object skuber {
...
+    case class Move(from: String, path: String) extends DirectionalOperation {
+      val op = "move"
+    }
+
+    case class Copy(from: String, path: String) extends DirectionalOperation {
+      val op = "copy"
+    }
+  }
+
+  sealed trait PatchStrategy
```
[the remark this answers is not in the document]
Comment (dtaniwaki) [reply client/src/main/scala/skuber/package.scala]: Thank you for the suggestion! I'll refactor it.

**Comment by dtaniwaki** [comment]
Thank you for the review! I refactored and improved the code. Could you check it again?

**Comment by doriordan** [comment]
LGTM - thanks!

**Pull request merged**